lib/cocoapods/generate/podfile_generator.rb

```diff
@@ -121,7 +121,41 @@ def podfile_for_spec(spec)
           end
 
           pod spec.name, **pod_options
+
+          # Implement local-sources option to set up dependencies to podspecs in the local filesystem.
+          next if generator.configuration.local_sources.empty?
+          generator.transitive_local_dependencies(spec, generator.configuration.local_sources).each do |dependency, podspec_file|
+            pod_options = generator.dependency_compilation_kwargs(dependency.name)
+            pod_options[:path] = if podspec_file[0] == '/' # absolute path
+                                   podspec_file
+                                 else
+                                   '../../' + podspec_file
```

Comment:
i dont get it, why '../../'?

Reply:
The workspace is generated two levels below the root in gen/{pod_name}